UltraStar Play/Assets/Scenes/Options/ContentOptions/ContentDownloadSceneController.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+using UniInject;
+using UniRx;
+using System.IO;
+using UnityEngine.Networking;
+using UnityEngine.UIElements;
+using ICSharpCode.SharpZipLib.Tar;
+using static ThreadPool;
+
+// Disable warning about fields that are never assigned, their values are injected.
+#pragma warning disable CS0649
+
+public class ContentDownloadSceneController : MonoBehaviour, INeedInjection
+{
+    public Text statusLabel;
+    public Text logText;
+    public Text downloadPath;
+    public Text fileSize;
+
+    public void StartDownload()
+    {
+        StartCoroutine(DownloadFile(downloadPath.text, PersistentTempPath()));
+    }
+
+    private string PersistentTempPath()
+    {
+        string path = Path.Combine(Application.persistentDataPath, "Temp");
+        //Create Directory if it does not exist
+        if (!Directory.Exists(path))
+        {
+            Directory.CreateDirectory(path);
+        }
+        return path;
+    }
+
+    private string PersistentSongsPath()
+    {
+        string path = Path.Combine(Application.persistentDataPath, "Songs");
+        //Create Directory if it does not exist
+        if (!Directory.Exists(path))
+        {
+            Directory.CreateDirectory(path);
+        }
+        return path;
+    }
+
+    private IEnumerator DownloadFile(string url, string targetFolder)
+    {
+        Uri uri = new Uri(url);
+        string filename = Path.GetFileName(uri.LocalPath);
+        string targetPath = Path.Combine(targetFolder, filename);
+
+        using (UnityWebRequest request = UnityWebRequest.Get(url))
+        {
+            AddToLog($"Starting file download for {uri}.");
+            DownloadHandlerFile downloadHandler = new DownloadHandlerFile(targetPath);
+            downloadHandler.removeFileOnAbort = true;
+            request.downloadHandler = downloadHandler;
+
+            StartCoroutine(TrackProgress(request, targetPath));
+
+            yield return request.SendWebRequest();
+
+            if (request.isNetworkError || request.isHttpError)
+            {
+                AddToLog("Error downloading the requested file!");
+            }
+            else
+            {
+                statusLabel.text = "100%";
+                AddToLog($"Download saved to: {targetPath}\r\n{request.error}");
+            }
```

Comment:
I have never seen Windows line ending \r\n beeing used in code. Why not use only \n?

Reply:
changed that.